Mathlib/CategoryTheory/Grothendieck.lean

```diff
@@ -149,6 +161,101 @@ def forget : Grothendieck F ⥤ C where
 
 end
 
+section
+
+variable {G : C ⥤ Cat}
+
+/-- The Grothendieck construction is functorial: a natural transformation `α : F ⟶ G` induces
+a functor `Grothendieck.map : Grothendieck F ⥤ Grothendieck G`.
+-/
+@[simps!]
```

Comment:
This `def` seems very slow, can you try to make it faster? Thanks!

Reply:
I agree this is a bit slow, it might help to split it up into separate defs `mapObj` and `mapMap`, and lemmas `map_map_id` and `map_map_comp`.

Reply:
It's only the `map_comp` field that's slow, it seems